src/components/ConditionalFilter/UI/BulkSelect.tsx

```diff
@@ -0,0 +1,120 @@
+import { DynamicMultiselect } from "@saleor/macaw-ui-next";
+import React, { useEffect, useRef, useState } from "react";
+
+import { FilterEventEmitter } from "./EventEmitter";
+import { BulkselectOperator, RightOperatorOption } from "./types";
+
+export interface BulkSelectProps {
+  selected: BulkselectOperator;
+  emitter: FilterEventEmitter;
+  index: number;
+  error: boolean;
+  helperText: string;
+  disabled: boolean;
+}
+
+const truncate = ({ label, ...rest }: RightOperatorOption) => ({
+  label: label.length > 15 ? label.slice(0, 8) + "..." : label,
+  ...rest,
+});
+
+const sanitize = (text: string) => text.replace(/\s/g, ",");
+const clean = (text: string) => text !== "";
+const toOption = (text: string) => ({
+  label: text,
+  value: text,
+  slug: text,
+});
+
+const BulkSelect = ({ selected, emitter, index, error, helperText, disabled }: BulkSelectProps) => {
```

Comment:
issue: I think you don't need an index here, when you expose events as props you can pass their index to the parent component